### ISSUE TITLE
Scenario comparison of PyPSA-Earth with third-party modelling frameworks

### ISSUE THREAD

**Issue by davide-f** [pz-max/pypsa-africa#622]
## Describe the feature you'd like to see

It would be appropriate to compare the modelling we obtain with PyPSA-Earth to other closed and open modelling frameworks.
The more studies, the better.

This task can be addressed also by beginners.

**Comment by stuenergy** [comment]
Hi Davide,
this seems like aninteresting task! I'd be up. Do we have a particular region we're looking at?
Best,
Stuart

**Comment by pz-max** [comment]
There was a discussion at GPST about this. In short, comparing expansion problems is difficult. One has to be clear about he constrains, renewable potential, included technologies, spatial resolution, temporal resolution, cost assumptions in detail etc. There is a need for creating benchmark cases that every model can be tested on 👍🏽

**Comment by ekatef** [comment]
@pz-max a very good point! As I recall from climate experiments (CMIP & Co), it's always a challenge to keep all the models more or less close to each other!

Would it be probably helpful for us to write-down kind of definition for mode-setup intended for usage in such multi-model inter-comparisons? 

I suspect, it might be very difficult to reproduce exactly settings of one energy model when user another one but we can at least try to be as close as possible and track differences

**Comment by davide-f** [comment]
Hello and nice for the interesting comments already.
I feel that we shall differentiate the goal of the comparison:
1. the comparison of modelling frameworks
2. the comparison of the final results

While performing any comparison of planning tools, it is implicit that we do both.
My feeling as to start, as a first analysis, it would be nice to run pypsa-earth for a country and compare the outputs with what we find in literature.
Clearly, input data will be different so results will be different but that's a start.
A draft comparison of input data may be done, if the third source states total demand and something alike.
As a starting point, also for the conference, I think just running a preliminary check on how far we are, eventually adjusting simple elements, e.g. scaling the total demand, may be appropriate.

A separate study may also do a more precise comparison same inputs - same outputs, this could be a separate study per-se.

**Comment by davide-f** [comment]
To start with, @stuenergy you may proceed as follows:
1. search the literature for existing planning studies for countries of relevance in Asia, South America or Africa, best if they use commercial tools. The more studies (1-3) we can compare to, the better
2. let us know, what options you have found so that we can agree on the country
3. Install and setup PyPSA-Earth for running that country. 
    a. If you have never run PyPSA-Earth, start with [the basic tutorial](https://pypsa-earth.readthedocs.io/en/latest/short_tutorial.html) for Nigeria
    b. once you get familiar, you may start using the standard configuration by deleting the original "config.yaml" and replacing it with a new copy of "config.default.yaml": this will enable you using 1-year long time series for any country of your choice; the short tutorial applies for only few countries. Also note that, to avoid data problems, I may recommend you to delete the folders "data/resources/results/networks" to avoid any issue
    c.  now you are ready to get started and use the tool for the region of your interest. Guidance is provided [here](https://pypsa-earth.readthedocs.io/en/latest/tutorial.html). Note, if you choose a region that is not in Africa, then you need to create your own cutout using the procedure described [here](https://pypsa-earth.readthedocs.io/en/latest/tutorial.html#build-the-custom-cutout) that requires registration and setup of the copernicus service (open to everyone after registration); links in the referenced page
    d. once the execution has been performed, you can use lots of notebooks in our [documentation repo (folder notebooks)](https://github.com/pypsa-meets-earth/documentation) that can help you plot images and do some analyses. A guide of the examples in that repo is reported [here](https://pypsa-earth.readthedocs.io/en/latest/notebooks.html)

For understanding and getting started, also during the learning using the short tutorial (point 3a) you may use the documentation notebooks (point 3.d) to investigate the data.

**Comment by davide-f** [comment]
@stuenergy is it ok for you?
To proceed, maybe you could search for some studies and post here what you found so that we could discuss over them.

**Comment by stuenergy** [comment]
Hey team,
hey @davide-f ,

thanks for your comments and inputs. Right now I'm not sure that I'm up to the task in the given time 😬 - especially because I am to prepare and hold the full "Energy Storage" lecture at DIT from scratch (starting in a week).
That being said I did do some research on studies that used energy system models in Southern Africa, because that is my current focus. I will not be able to look into other regions. That is not agreeable with my current tasks.

I totally understand if you'd like to look into other regions. I guess that was the whole point of this excerise - at least how I get it. In that case I'd, however, be willing to take over other tasks.

What are your thoughts?
Do you have someone else in the pipeline for this task? Then it might be a safer option to go with them.

If I were to continue with some Southern African country: **We do need a study based on a closed-source energy system model, right?** Cause most that I have found were open-source.

Btw: Changed my account email address to my business address. Now my responses shouldn't take that long anymore. Sorry for that!

**Comment by davide-f** [comment]
Hello @stuenergy :)

Currently no one else has showed up for this task.
Having a closed source one would be a nice-have but not mandatory if we cannot find it

**Comment by davide-f** [comment]
Examples of studies using:

Open source studies:


Closed-source tools:
- (plexos) Lesotho - https://doi.org/10.1016/j.enpol.2018.06.003
- (plexos) thesis on South Africa - http://crses.sun.ac.za/old/files/research/completed-research/eppei/MSklarchik.pdf

**Comment by stuenergy** [comment]
Hey @martacki,
@davide-f mentioned on Monday that you might have some potential studies up your sleeve that are worth comparing our results to. Would be great if you could post what you have here. :)
Best,
Stuart

**Comment by stuenergy** [comment]
@davide-f 

Putting out some questions while I go:
- LEAP is not quite the tool that we'd like to compare our results with cause it's an accounting model, right?
- Which cut-off criteria should we go for regarding when the study was made? (2013 is too old I guess)
- I guess we do not want to compete with OSeMOSYS (cause they are our friends)?

Found these studies in my Southern Africa stash:
- 12 countries, Southern Africa, using **GridPath**: https://linkinghub.elsevier.com/retrieve/pii/S254243512200304X
- 12 countries, Southern Africa, using **MESSAGE**: https://www.irena.org/publications/2013/Jun/Southern-African-Power-Pool-Planning-and-Prospects-for-Renewable-Energy
- 12 countries, Southern Africa, using **PLEXOS**: https://www.sapp.co.zw/sites/default/files/SAPP%20Pool%20Plan%202017%20Main%20Volume_0.pdf
- Democratic Republic of the Congo, using **GridPath**: https://iopscience.iop.org/article/10.1088/1748-9326/aabf60
- Angola, using **HOMER Pro** (closed-source, but rather Microgrids, right?): https://ieeexplore.ieee.org/document/9543099
- South Africa, using **MATLAB & MOSEK v8**: https://doi.org/10.1016/j.solener.2019.09.039
- Tanzania, **OSeMOSYS**: https://doi.org/10.1016/j.esr.2020.100614

For me, the ones looking into the SAPP countries are super relevant - probably too complex for our paper though?

Btw: The Southern African Power Pool (SAPP) uses PLEXOS for their energy pool plan. This could be an interesting angle - to show them that they could do it cheaper by using PyPSA-Earth! What do you think?

**Comment by davide-f** [comment]
Wow Stuart, cool overview!
The Sapp would be amazing! Maybe an overkill for the conference but if you are interested it would be amazing :)
We could do that, a run of Sapp should be easy to do and we have all instruments for validation.
Also the comparison with plexus would be very nice to have.
For me sapp can be the case study :)

**Comment by stuenergy** [comment]
Sounds good, that would be absolutely perfect and easy to explain to my advisors! 😂

For an overview I identified these studies that use PLEXOS:
- South Korea: https://www.mdpi.com/1996-1073/15/9/3310
- Australia (bit old): https://doi.org/10.1016/j.renene.2013.05.009
- Malaysian Borneo: https://doi.org/10.1016/j.esr.2015.07.001
- Lesotho (the one you already found): https://doi.org/10.1016/j.enpol.2018.06.003

However, I'll go for the SAPP then cause that would really make sense - gotta do that anyway! 😊

**Comment by martacki** [comment]
> Hey @martacki, @davide-f mentioned on Monday that you might have some potential studies up your sleeve that are worth comparing our results to. Would be great if you could post what you have here. :) Best, Stuart

@stuenergy 

Hey, here's a European study comparing wind, hydro, pv feed-in from PLEXOS to ENTSO-X reported data https://reader.elsevier.com/reader/sd/pii/S0142061520309534?token=6C0C2C986E31D8362AF6E22D1EDFF9ADA682257514E045E13A78E06834A43CA575AF70515AB611C0B7F15D6F6D780A21&originRegion=eu-west-1&originCreation=20230317141853

Here's one non-comparison modeling results (4 scenarios) for selected States of the US, powered by NREL https://www.nrel.gov/docs/fy13osti/55588.pdf. They have a lot of information about input data.

Would be somewhat cool if we could show that we can model that with pypsa-earth.

However, I'm not sure how useful these resources are, because its Europe or US and not Asia/Africa. Sorry for that. But seems like Stuart has already found some useful info for the intended regions.

**Comment by martacki** [comment]
> However, I'll go for the SAPP then cause that would really make sense - gotta do that anyway! blush

If you want to go with SAPP, there is super much nice visualisation on the input data. They also seem to have published a lot of cost assumptions etc. (Table 6), which is nice... but what about a ppl database (in the pypsa-earth model)? Do we have that already, or only the grid data? In any case, the study seems to make extremely many manual model adaptations to include (other) planning results (eg. page 43). I think it might be a looooot of work adapting the network topology... It might be worth scanning first what is already there in the model, what information is missing but can be easily acquired from the study, and then check what data you would need to acquire from other sources. This gives you insights if the study is doable (within the paper deadline). For this you'd first need to check which outputs you want to generate: There is simply too many in the study for a conference paper. I'd suggest to pick only a few. For example import/exporters (page 51/2), Energy balances (page 49), and/or shares (page 50). At least this would be needed for the Plexos comparison.

**Comment by davide-f** [comment]
> > However, I'll go for the SAPP then cause that would really make sense - gotta do that anyway! blush
> 
> If you want to go with SAPP, there is super much nice visualisation on the input data. They also seem to have published a lot of cost assumptions etc. (Table 6), which is nice... but what about a ppl database (in the pypsa-earth model)? Do we have that already, or only the grid data? In any case, the study seems to make extremely many manual model adaptations to include (other) planning results (eg. page 43). I think it might be a looooot of work adapting the network topology... It might be worth scanning first what is already there in the model, what information is missing but can be easily acquired from the study, and then check what data you would need to acquire from other sources. This gives you insights if the study is doable (within the paper deadline). For this you'd first need to check which outputs you want to generate: There is simply too many in the study for a conference paper. I'd suggest to pick only a few. For example import/exporters (page 51/2), Energy balances (page 49), and/or shares (page 50). At least this would be needed for the Plexos comparison.

Thanks @martacki for the comments.
My feeling is that if we combine all the ideas we may write 3 papers instead of 1 XD
A possible approach may be to use pypsa-earth "as-is" the most the possible.
Personally, I'd be very curious to check what we get with our default configuration vs third-party analysis.
Errors may (or may nor) be large, but showing that could be already a good output for the conference.
Obviously we will compare different things aiming at representing the same thing but a preliminary comparison for the conference should be ok; in a future journal paper we could improve that more and add all the details in the comparison.

What do you think?

@stuenergy, anyway, I think the starting point should be to define the list of SAPP countries and run the workflow using the standard configurations. We may change the number of buses with a "large" number.
We may test 2/3 configurations to see what happens. I think that the first run may be a run using the number of clustering nodes matching the ones used in the study.

In short the procedure should be:
1. copy config.default.yaml as config.yaml
2. change countries: [ {list of SAPP countries} ]
3. change the value of scenario... clusters on top of the script to match the value of the study
4. run the model
5. doing some plotting using scripts in our [documentation/notebooks/validation](https://github.com/pypsa-meets-earth/documentation/tree/main/notebooks/validation) and post here some of them with some considerations regarding the study on SAPP that you found

What do you think?

**Comment by stuenergy** [comment]
Thanks @martacki and @davide-f for your valuable insights! I will do as you propose Davide. I will reach out as soon as I get into troubles. :D

**Comment by davide-f** [comment]
Super :) Would be cool to have some raw/preliminary results/images next week.
I remember the deadline April 14th...

**Comment by stuenergy** [comment]
Hi @davide-f, I'm currently running the model with default settings for the clusters (55). I could not find any clues on how many clusters are used in the study. Is there another way to find out or approximate this value?

**Comment by stuenergy** [comment]
@davide-f, @pz-max
Hmm, I'm getting an error:
![grafik](https://user-images.githubusercontent.com/87220279/227492382-06001f1d-75d1-4e90-91c6-e40325d09197.png)
with:
![grafik](https://user-images.githubusercontent.com/87220279/227492558-6ad1ee69-9104-4149-af7b-85888729ccce.png)
I'm rather confused as I did not choose to look at Morocco. I guess it's all downloaded anyways?
How do I solve this issue?
As soon as it's solved I'll have to start over, right? (clean all data and run the model again etc.)

**Comment by stuenergy** [comment]
`[Fri Mar 24 09:54:01 2023]
INFO:snakemake.logging:[Fri Mar 24 09:54:01 2023]
Error in rule build_shapes:
    jobid: 12
    input: data/eez/eez_v11.gpkg
    output: resources/shapes/country_shapes.geojson, resources/shapes/offshore_shapes.geojson, resources/shapes/africa_shape.geojson, resources/shapes/gadm_shapes.geojson
    log: logs/build_shapes.log (check log file(s) for error details)

ERROR:snakemake.logging:Error in rule build_shapes:
    jobid: 12
    input: data/eez/eez_v11.gpkg
    output: resources/shapes/country_shapes.geojson, resources/shapes/offshore_shapes.geojson, resources/shapes/africa_shape.geojson, resources/shapes/gadm_shapes.geojson
    log: logs/build_shapes.log (check log file(s) for error details)`

**Comment by ekatef** [comment]
Hey @stuenergy, could you please share the whole listing of the error? Usually there is a long list of Snakemake complains in the console

As for Morocco could you please specify which config parameters in `config.yaml` you are using, and in particular what is `countries` parameter? Morocco shouldn't be loaded if it's not specified explicitly

Sorry for jumping in :) I suspect Max and Davide could be not quite available at the moment

**Comment by stuenergy** [comment]
Thanks so much @ekatef! Now that you say it... There is a bunch of complaints that I did not recognise cause they're not flashy red...
![grafik](https://user-images.githubusercontent.com/87220279/227503602-b2464c13-4e8e-4f40-9504-84a15a06745b.png)
![grafik](https://user-images.githubusercontent.com/87220279/227503804-940b35ff-ee39-4e32-83b9-fdaf61de9e41.png)

Regarding config parameters:
![grafik](https://user-images.githubusercontent.com/87220279/227504633-0472234f-e7d6-45c7-9dff-badab258fbe2.png)
![grafik](https://user-images.githubusercontent.com/87220279/227504739-d7718ef4-66d9-4e7d-ae30-dbaa2def59c6.png)

Thanks again! :)

**Comment by pz-max** [comment]
@stuenergy , the lines below Traceback are most important. Reading theTraceback meesage it seems like a weird error in  `build_shapes.py` and in particular some kind of GADM function. Can you debug this especially analysing the URL and the role of Morroco (MAR)? (see hackathon material and youtube videos on how to debug)

**Comment by ekatef** [comment]
Thanks a lot for the detailed report!

The error seems to be caused by some troubles in connecting to GADM server. But the server looks alright at the time being (CI is green and I'm able to load gadm data locally). Which gives some hope that the troubles you are facing can just resolve by themselves

Configuration settings look perfectly correct for the case you are investigating, why Morocco is loaded is a bit of a mystery... Could there be any troubles with mixing-up folders/different config.yaml versions?

**Comment by ekatef** [comment]
@stuenergy, both errors (with gadm and loading a non-specified country) were not seen before. So, right now you are contributing into making the model more stable and usable ;)

**Comment by stuenergy** [comment]
Hooraaaaay! 😃😂
Thanks for your feedback @ekatef and @pz-max, I'll look at it more closely after having devoured my Shawarma. 😉

**Comment by davide-f** [comment]
Hello @stuenergy,
Nice that you are moving forward :D

Some comments:
- it sounds weird that you are downloading morocco; make sure you save the file config.yaml before running snakemake.
   In the screenshop there is the "white dot" next to the name of the file. That means that the file is not saved and hence snakemake is not loading what you see in the screen.
- that error is a server issue: a feeling could be that you run the config for Africa (maybe the old configuration before your changes) and then the workflow started for a long list of countries and maybe there are some limitations on the number of downloads (or something alike)

**Comment by stuenergy** [comment]
Thanks @davide-f, @pz-max and @ekatef for your support!
@ekatef: You were right about the VPN... 👍

However, I have just come back home to this:
![grafik](https://user-images.githubusercontent.com/87220279/227614344-ae564e64-75bf-464b-8c48-b4e598d45fd1.png)
I guess that is how it goes. :D Won't be able to fix this on the weekend as I'm attending a workshop tomorrow and celebrating my sisters 30th birthday on Sunday - but I'll be bakc at it on Monday. 
I wish you all a great weekend!

**Comment by davide-f** [comment]
Hello!
The error message is above, similarly to before.
Unfortunately with this, we cannot help :(

**Comment by stuenergy** [comment]
Hey @davide-f, @ekatef & @martacki,

I hope you had nice holidays!
I have actually managed to get as far as to the point when the solver starts:
`INFO:pypsa.linopf:Solve linear problem using Glpk solver`

<img width="699" alt="2023-04-11 10_07_01-terminal-window" src="https://user-images.githubusercontent.com/87220279/231101231-cb71aef3-672c-4fec-b0c2-adada643a4a4.png">

However, it has been like that since Thursday around noon and nothing has happened since (I left the computer on until now). Now I am wondering what the issue is: Does it just take that long, or has my virtual machine hung itself up? (I can tell that it is requiring a lot of memory - screenshots of task manager attached).

<img width="334" alt="2023-04-11 10_09_07-Task Manager" src="https://user-images.githubusercontent.com/87220279/231101294-8262a7a6-0c9e-474d-a56d-1f2ab0754591.png">

<img width="260" alt="2023-04-11 10_09_42-Task Manager" src="https://user-images.githubusercontent.com/87220279/231101310-c65f2f63-7437-4cb3-a335-6f14620630e7.png">

I would really appreciate your views on this. Also: If it has hung itself up: How best do I proceed? Will I have to run the complete model again, or can I restart with the individual rules? Because as far as I understand, the required data is there already.

You'll find the relevant config file settings attached, as well as some task manager snippets and the last few lines from my terminal window.

<img width="702" alt="2023-04-11 10_07_37-config yaml - pypsa-earth-project  WSL_ Ubuntu  - Visual Studio Code  Administra" src="https://user-images.githubusercontent.com/87220279/231101486-c9143758-6382-4d8b-b22c-2908fc9350ad.png">

<img width="702" alt="2023-04-11 10_08_10-config yaml - pypsa-earth-project  WSL_ Ubuntu  - Visual Studio Code  Administra" src="https://user-images.githubusercontent.com/87220279/231101523-6fd6fcfc-9a00-4dbd-8dcf-793c19e7acc8.png">

Thanks so much in advance!
Best,
Stuart

**Comment by martacki** [comment]
Hey @stuenergy 
It's not really normal that the solving takes that long for a 28 node network and 3h resolution in time. (BTW the 3H setting has worse results than e.g. 2920SEG if you have renewables in the network).
However, I see you're not using gurobi but glpk, that could result in slower solving times. But it does not justify such a large overhead. Therefore, I think  your machine is frozen. You can start anew from solve_network (no need to execute anything before). I'd suggest to solve lower resolution first just to see if it solves, e.g. elec_s_12_ec_lcopt_Co2L-25H or even elec_s_12_ec_lcopt_Co2L-1000H, this is something like 9 snapshots. For the lower-resolved testing you of course have to re-run cluster_network + rest. The solving then should be done faster than a minute. If it does not work, you probably have a different problem...
 
Also, if possible, I'd try using gurobi. You can get a free academic license from their server. It's just the best current solver we have.

(Problem could also be pyomo, but I think you're not using pyomo, right? Judging by the logging info...)

**Comment by stuenergy** [comment]
Hey @martacki,
thanks a lot for your helpful response. I have narrowed down the problem to Visual Studio Code continuously disconnecting from my WSL-Ubuntu. It just keeps freezing - sometimes showing me according messages and other times just ..freezing...
I will call our IT today cause this is not a feasible setup and they actually wanted to set up a server for me which would be ideal.

![2023-04-13 08_27_47-config yaml - pypsa-earth-project  WSL_ Ubuntu  - Visual Studio Code  Administra](https://user-images.githubusercontent.com/87220279/231677398-7e46a661-238d-400a-ac1d-8b525abbd8ce.png)

@davide-f: I know the deadline is soon and I am really sorry that it is taking longer than expected which is why I am giving you a heads up here. I am not sure how confident I am that I have enough time to meet the deadline.
**When do you need a first draft?**

I just want to communicate the issues I am having very clearly to avoid that I blow the whole thing unexpectedly. Gonna call IT now.

**Comment by martacki** [comment]
Hi @stuenergy @davide-f 
We can cooperate here. If you find a smooth way to send me the ready-to-solve networks i.e. elec_s_28_ec_lcopt_Co2L-3H.nc files, I can run the solver for you. Ideadlly before sending, you could change 3H->2920SEG. Also, can you tell me what is in your config for Co2L, and what clustering do you use to generate 28 nodes?
The deadline is very soon and we need results. I didn't write anything on this part yet because I didnt want to waste time in case we can't finish.

What do you think Davide?

**Comment by stuenergy** [comment]
Thanks for your offer @martacki!
For Co2L I am using the default settings as suggested by @davide-f, for clusters I first set 28 and in the next go 12 as recommended. Not sure if I should use a value in between?
Also, I didn't find 2920SEG in the documentation. What exactly does it do / stand for?

I am starting to believe that this issue is not as beginners-friendly as I had hoped for. But that may just be my weird setup..

**Comment by davide-f** [comment]
@stuenergy regarding disconnections, I'd recommend to run the code under the "screen" command.
That works under linux environment and enables processes to continue even without the user being connected.

This means that you could launch the screen "screen -S name"
then you can run within that screen whatever and if you want to disconnect you can do: ctrl+a d
The process will continue running.
When you want to connect again, you can do "screen -r name" and you will link back to the session and see the updates

**Comment by stuenergy** [comment]
Hi @davide-f, as discussed on Friday we might need to opt for plan B. The solving on my laptop simply won't do anything (although it did create elec_s_25_ec_lcopt_Co2L-2920SEG.nc) and there are some major issues with my university profile in combination with the remote computer I'm supposed to use - above all a lack of time to sort things out during working hours (so with somebody from university IT available). Therefore, you'll find the requested files attached (config-file and elec_s_25_ec_lcopt_Co2L-2920SEG.nc).

[https://drive.google.com/drive/folders/130T-T7NlQn9q6h_c1dXuyloHWUuppuKs?usp=share_link](url)

Please let me know how we will proceed from here and how I can be of maximum benefit in the further process.

**Comment by davide-f** [comment]
@stuenergy please, find in the same folder the optimized network.
I used gurobi to optimize it. Please, copy the folder "results" as-is into the pypsa-earth and that should do the trick.

You can use that optimized network to produce outputs using the documentation folder :)
We are there :D

**Comment by stuenergy** [comment]
You're a fantastic man @davide-f!
Thanks a lot for this. I'm guiding my students to a field trip to beautiful Walchensee and its storage power plant full day tomorrow, but Friday I'll have 24 hours to get it done! :D

**Comment by stuenergy** [comment]
Dear @davide-f,
Alright. So, I have managed to get some results. On some of the Jupyter-Notebooks I haven't been successful yet.

OSM clean Substations and lines:
![output](https://user-images.githubusercontent.com/87220279/233669714-37316377-c281-4359-a41c-c68ed3186e17.png)

Demand map:
![output1](https://user-images.githubusercontent.com/87220279/233669993-82a596a1-70e2-4c19-b033-09c60c0a0e09.png)

Clustering visualisation:
![grafik](https://user-images.githubusercontent.com/87220279/233670193-d6016775-b3bb-4e5d-8856-f4a1d561675b.png)

With the notebook: "nice_plotting_for_presentations" I am running into this error:
![grafik](https://user-images.githubusercontent.com/87220279/233671366-f29b6470-2325-4e5a-b54f-157ea2d1c1a6.png)
And indeed:  The name 'create_country_list' from 'download_osm_data' (/home/stuart/pypsa-earth-project/pypsa-earth/scripts/download_osm_data.py) does not appear in that file. This must have changed at some point?
**What is the variable actually called that I am looking for in 'download_osm_data'?**

With the notebook: "landuse-availability" I would like to plot the entire SAPP region and not just a single country. I have tried some variations - none worked.
**Will I have to define several countries as one "country (or region)" at some point for this to work?**

I am really not optimistic that I will be able to deliver something useful in time. I am totally blocked on Monday and Tuesday and also a good portion of Wednesday. I am really giving it my best. That might not be enough.
**Do you have any suggestions of what I should compare to?**
OR:
**Is this actually a sensible basis to compare with?**

**Comment by davide-f** [comment]
Nice! @stuenergy 

I think it would be nice to reproduce a table or some results that the other article obtained.
Maybe a demand plot with bars similarly to the one prepared by Leon in the overleaf.
Moreover, the plot of the network compared to a plot from the report may be interesting.

Could you also prepare some images or relevant data from the report to make the comparison?

Happy to discuss later or also in the weekend

**Comment by davide-f** [comment]
The error you are experiencing is due to the fact that the create country list has been moved to the helpers.

Have you named the parent folder of pypsa-earth "pypsa-earth-project"?
If not, please do so

**Comment by stuenergy** [comment]
Hi @davide-f,
hi @martacki,

again: thanks so much for your support so far!!!! 🤗
I like the idea Davide. The only nice-looking thing that I have managed to derive so far has been the lines and substations plot. This we could at least compare to the according map from the SAPP report:
![grafik](https://user-images.githubusercontent.com/87220279/234283254-293d97c0-ec15-41b6-89e5-cc9fa1701fff.png)

I have also managed to find SAPP's demand forecast to 2040 that we could compare our results to. However, I have no clue how to derive these results from our model and would need help doing so:
![grafik](https://user-images.githubusercontent.com/87220279/234283651-cb0d053f-c1a7-438e-adf4-99659214f511.png)
Once I have results (either SAPP total - or - as they did in the SAPP report: South Africa + Rest of SAPP) I am happy to create a diagram just like Leons although I do not know which tool he used to create the diagram. Doesn't look like excel... 🙈

Something else that I have managed to derive is this demand map that looks like crap - especially compared to the SAPP demand map. So no good using this I'd say:
![grafik](https://user-images.githubusercontent.com/87220279/234284497-2add0fbe-3a27-4d92-8f21-dc123cecbb33.png)

Also, I have written something in our paper and would add some general discussion as soon as I have the demand results to compare with. I also tried to describe the section as a first glimpse into a model comparison as obviously that's not quite what this is. It's more an accident. I literally feel like a toddler who has been sat down in front of some super-delicate artwork and is now looking at the mess he made - rather unsure whether his parents will approve. 🤣

Speak you soon!

**Comment by stuenergy** [comment]
@davide-f : I am getting this error message when trying the `snakemake -j 1 networks/elec_s_25_ec_lv1.0_Co2L-2920SEG.nc`:
![grafik](https://user-images.githubusercontent.com/87220279/234311966-ce5f0c24-b848-489b-b194-94eebceee8c4.png)
I have no idea where to even start.

**Comment by martacki** [comment]
@stuenergy seems to have to do with the co2limit/co2base that we have adapted in the meeting. try 1e+9?

**Comment by stuenergy** [comment]
That wasn't it I'm afraid:
![grafik](https://user-images.githubusercontent.com/87220279/234326610-a9622442-97bd-4ce6-a3ba-61b6d28390e1.png)
@martacki @davide-f

**Comment by martacki** [comment]
Hm, still either `Nyears` or `annual_emissions` seems to be in the wrong dataformat. try to reset co2limit/base to original values? just to see if it works

**Comment by stuenergy** [comment]
@martacki 
Figured it out! Apparently it needs a 1.0e+10 instead of 1e+10 ...
I uploaded the according files to the folder "Updated files". (incl. config file)

Here: https://drive.google.com/drive/folders/130T-T7NlQn9q6h_c1dXuyloHWUuppuKs?usp=share_link

**Comment by stuenergy** [comment]
@martacki

1. **Should I list reported capacities of all energy carriers?**
2. **Can't we just assume that SAPP uses the best available data in their report?** They basically got it directly from the utilities.
3. **With reported generation we are looking for current values, right?** Something in the range of 2020.
4. **Do you have any hints where best to look literature-wise?**
    --> for "reported capacities" (except the SAPP report itself)
    --> for "reported generation in TWh"

Last question maybe also @davide-f or @ekatef ?

**Comment by davide-f** [comment]
@stuenergy Regarding 4, I recommend checking [this PR](https://github.com/pypsa-meets-earth/documentation/pull/48) by Leon.
He used an IRENA and IEA datasets for such purpose.

**Comment by davide-f** [comment]
@stuenergy @martacki  Network optimized and shared in updated files under results/networks folder

**Comment by stuenergy** [comment]
@davide-f @martacki
So, the SAPP report (and its annexes) does have great information on demand forecast which they call "energy sent out" - even a time-series and we could pick out the 2030 value as well for direct comparison with PyPSA-Earth:
![grafik](https://user-images.githubusercontent.com/87220279/234499745-2d3bf1a3-4c94-41b2-9853-cfbef698278a.png)
(One for the Base-Scenario and one for the Low-Scenario)

They do not have a strong focus on total generation per energy carrier. They rather look into generation planning / future additions to the generation mix (and transmission) with a strong focus on specified planned power plants and their economic feasibility. However, I did find this in the report:
![grafik](https://user-images.githubusercontent.com/87220279/234499450-e11ce471-67f4-4c52-9278-91f0620a462a.png)
Not sure this is helpful for us though, as they do not have any tables or values written anywhere.

So it basically looks like we only really have the demand forecast [TWh] to properly compare to. We could however also compare to the "sent out peak demand" [GW]. They have similar tables as with the demand forecasts - although not fully complete because they did make a mistake in one of the tables and inserted the [TWh]-values instead.

**Comment by martacki** [comment]
there must be official feed-in numbers/carrier for SAPP region. without this information there is no point in this section.  I dont have time to read all the material at this stage. @stuenergy

**Comment by stuenergy** [comment]
@martacki: Found something outside the report! I'll add it to another table.

**Comment by davide-f** [comment]
@stuenergy @martacki Many thanks for the contirbutions.
I've done another round of revision and it seems we are converging.

@stuenergy could you add all *.nc files in the networks folder of your model?
I'd like to check the elec.nc file for the demand stats.

Could you please check the document tomorrow even morning and let me know?
Please, write me on discord and/or on the mobile for quick answers.

During the next days I'll do the details to then submit by the deadline

**Comment by martacki** [comment]
i think I still have to add TABLE 2 + Table 1/2 descriptions. The installed capacities are a bit off, particularly Gas, but we should just be honest and say "look there is still potential for improvement, but note that this is the first package and it is still very accurate in many technology assumptions"

**Comment by davide-f** [comment]
Yeah and agree.
A conference paper is meant to be a preliminary (but good) analysis.
It also makes sense to have different values, we just need to word it in a good way

**Comment by stuenergy** [comment]
Thanks a lot @davide-f and @martacki.
I just added some descriptions for tables 1 and 2 - considering your thoughts @martacki.
Feel free to revise @davide-f!

**Comment by davide-f** [comment]
Awesome!
On my side, I'll be able to do something tonight.

**Comment by davide-f** [comment]
Many thanks @stuenergy for the support! :)
I think this PR can be close as a preliminary comparison has been done.
We can reopen it if necessary later on :) Many thanks martha and everyone for the support!
